### DISCUSSION TITLE
IC loss for the 2D Allen-Cahn version does not change after first iteration

### DISCUSSION THREAD

**Discussion by jannesvpoppelen** [lululxvi/deepxde#1473] [Q&A]
Hello!

I tried implementing the 2D Allen-Cahn equation to get familiar with the library. I am aware that there is a 1D example available, however, that implements the boundary and initial conditions in a different way than I do, and I don't understand how to rewrite mine to be consistent with the example. I think I have everything set up correctly, but when I try to train the network, the IC loss does not change at all. I have taken a look at the FAQ ( I failed to train the network or get the right solution, e.g., large training loss, unbalanced losses.), and tried some suggestions there, such as changing the weight, adding more num_initial, changing the time interval, and all of these did not seem to resolve the issue.

The exact problem I tried solving is

$$\frac{\partial u}{\partial t} = k (\frac{\partial^2u}{\partial x^2} + \frac{\partial^2u}{\partial y^2}) - \frac{1}{\epsilon^2} u(u^2-1)$$

$$u(x, y=0, t) = u(x, y=1, t)$$

$$u(x=0, y, t) = u(x=1, y, t)$$

$$u(x,y, t=0) = \sin(4\pi x)\cos(4 \pi y)$$

Any help would be greatly appreciated.

Additionally, I would like to use JAX as the backend for the library, while right now I am using Tensorflow. There do not exist any examples that use JAX, and the syntax seems to be slightly different when compared, so if I can get some pointers on how to translate this code using the JAX backend, I would also appreciate that!

Here is my code:

```python
"""
Implementation of 2D Allen-Cahn equation

d/dt u - k * (d^2/dx^2 + d^2/dy^2) u + 1/eps^2 * u * (u^2 - 1) = 0

Boundary conditions are periodic, and u(x,y,t=0) = sin(4 pi x) * cos(4 pi y)
"""
import deepxde as dde
import numpy as np
from tensorflow import sin, cos

# Global parameters
eps = 1
k = 0.0001  # μm^2/s
iters = 40000  # number of iterations for training

Omega = dde.geometry.geometry_2d.Rectangle((0, 0), (1, 1))  # μm
T = dde.geometry.TimeDomain(0, 1) # s
geometry = dde.geometry.GeometryXTime(Omega, T)


# Specify PDE and BCs
def pde(x, u):  # x = [x, y, t], u = [u]
    u_t = dde.grad.jacobian(u, x, i=0, j=2)
    u_xx = dde.grad.hessian(u, x, i=0, j=0)
    u_yy = dde.grad.hessian(u, x, i=1, j=1)
    return u_t - k * (u_xx + u_yy) + 1 / (eps ** 2) * u * (u ** 2 * - 1)


def h_boundary(x, on_boundary):
    return on_boundary and (np.isclose(x[0], 0) or np.isclose(x[0], 1))


def v_boundary(x, on_boundary):
    return on_boundary and (np.isclose(x[1], 0) or np.isclose(x[1], 1))


def ic(x):
    return sin(4 * np.pi * x[:, 0:1]) * cos(4 * np.pi * x[:, 1:2])


BC1 = dde.PeriodicBC(geometry, 0, h_boundary)
BC2 = dde.PeriodicBC(geometry, 1, v_boundary)
IC = dde.IC(geometry, ic, lambda _, on_initial: on_initial)

data = dde.data.TimePDE(geometry, pde, [BC1, BC2, IC], num_domain=8000, num_boundary=100, num_initial=2000, num_test=2000)

# Define network architecture
layers = [2] + [32] * 3 + [1]
activation = "tanh"
initializer = "Glorot uniform"
network = dde.nn.tensorflow.fnn.FNN(layers, activation, initializer)

# Define model
model = dde.Model(data, network)
model.compile("adam", lr=0.001, loss_weights=[1, 1, 1, 50])
model.train(iterations=iters, display_every=100)
model.compile("L-BFGS-B")
losshistory, train_state = model.train()
dde.saveplot(losshistory, train_state, issave=False, isplot=True)
```

**Comment by lululxvi** [comment]
Use `np.sin` and `np.cos` in `ic`.

**Reply by jannesvpoppelen** [reply]
I tried that before, and unfortunately that did not seem to work either

**Reply by ANTAGNIST** [reply]
just change the 2 to 3 in layer input 
```layers = [3] + [32] * 3 + [1]```
